File: src/main/scala/com/incra/model/Facility.scala
package com.incra.model

import scala.slick.driver.MySQLDriver.simple._

/**
 * Definition of the Facility entity
 *
 * @author Jeff Risberg
 * @since 08/12/2015
 */
case class Facility(id: Option[Long], name: String, lat: Double, lng: Double) extends Entity[Long]

class FacilityTable(tag: Tag) extends Table[Facility](tag, "FACILITY") {

  def id = column[Long]("ID", O.PrimaryKey, O.AutoInc)

  def name = column[String]("NAME")

  def lat = column[Double]("LAT")

  def lng = column[Double]("LNG")

  // Every table needs a * projection with the same type as the table's type parameter
  def * = (id.?, name, lat, lng) <>(Facility.tupled, Facility.unapply _)
}

File: src/main/java/com/incra/models/Facility.java
package com.incra.models;

import com.incra.database.AbstractDatedDatabaseItem;

import javax.persistence.*;

/**
 * Piece of a page
 *
 * @author Jeff Risberg
 * @since February 2014
 */
@Entity
@Table(name = "facility")
public class Facility extends AbstractDatedDatabaseItem {

    @Basic
    private String title;

    @Basic
    protected Double lat;

    @Basic
    protected Double lng;

    public Facility() {
    }

    @Override
    public String toString() {
        StringBuffer sb = new StringBuffer();

        sb.append("Facility[title=");
        sb.append(title);
        sb.append("]");

        return sb.toString();
    }
}